scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,476 @@
+    master_ip: The IP of the head node.
+    status_dict: A dictionary to record the status of the executed process.
+    keyname: A string containing the deployment's keyname.
+  """
+  try:
+    datastore.create_table(DATASTORE_METADATA_TABLE, DATASTORE_METADATA_SCHEMA)
+    cell_values = {}
+    cell_values[VERSION_INFO_KEY] = {
+      DATASTORE_METADATA_SCHEMA[0]: VERSION_1
+    }
+    time.sleep(5)
+    datastore.batch_put_entity(DATASTORE_METADATA_TABLE, [VERSION_INFO_KEY],
+      DATASTORE_METADATA_SCHEMA, cell_values)
+  except AppScaleDBConnectionError as conn_error:
+    logging.error("Error storing the datastore version: {}".format(conn_error))
+    status_dict[STORE_DATASTORE_VERSION] = str(conn_error)
+    close_connections(zookeeper, db_ips, zk_ips, master_ip, status_dict, keyname)
+    return
+
+def drop_journal_table(datastore, zookeeper,db_ips, zk_ips, master_ip, status_dict, keyname):
+  """ Drop JOURNAL_TABLE.
+  Args:
+    datastore: A reference to the batch datastore interface.
+    zookeeper: A reference to ZKTransaction, which communicates with
+      ZooKeeper on the given host.
+    db_ips: A list of database node IPs to stop Cassandra on, in case of error.
+    zk_ips: A list of zookeeper node IPs to stop ZooKeeper on, in case of error.
+    master_ip: The IP of the head node.
+    status_dict: A dictionary to record the status of the executed process.
+    keyname: A string containing the deployment's keyname.
+  """
+  try:
+    datastore.delete_table(JOURNAL_TABLE)
+  except AppScaleDBConnectionError as conn_error:
+    logging.error("Error deleting the JOURNAL_TABLE: {}".format(conn_error))
+    status_dict[DELETE_JOURNAL_TABLE] = str(conn_error)
+    close_connections(zookeeper, db_ips, zk_ips, master_ip, status_dict, keyname)
+    return
+
+def all_services_started(status_dict):
+  """ Loops through the values of the status dictionary to check the status
+  of ZooKeeper and Cassandra started on their nodes.
+  Args:
+    status_dict: A dictionary to record the status of the executed process.
+  """
+  for value in status_dict.values():
+    if value == FAILURE:
+      return False
+  return True
+
+def run_datastore_upgrade(zk_ips, db_ips, master_ip, status_dict, keyname):
+  """ Runs the data upgrade process of fetching, validating and updating data
+  within ZooKeeper & Cassandra.
+  Args:
+    zk_ips: A list of ZooKeeper node IPs.
+    db_ips: A list of database node IPs.
+    status_dict: A dictionary to record the status of the executed process.
+    keyname: A string containing the deployment's keyname.
+  """
+  # This datastore upgrade script is to be run offline, so make sure
+  # appscale is not up while running this script.
+  ensure_app_is_not_running()
+
+  # Start Cassandra and ZooKeeper.
+  start_cassandra(status_dict, db_ips, master_ip, keyname)
+  start_zookeeper(status_dict, zk_ips, master_ip, keyname)
+
+
+  # Sleep time for Cassandra and ZooKeeper to be started.
+  time.sleep(SLEEP_TIME)
```

Comment:
Instead of sleeping an arbitrary length here, I think it would be more reliable to check Cassandra's readiness directly:
```
while subprocess.call([NODE_TOOL, 'status']) != 0:
  time.sleep(1)
```
You can import `NODE_TOOL` from `cassandra_interface`.
For zookeeper, you can do the same with `/usr/share/zookeeper/bin/zkServer.sh status`.